### PR TITLE
[backend] POST endpoint

### PR DESCRIPTION
**Pull request by shayna-k**
Added:
- functionality to post a poem
- test code in index.html to see it works
- placeholders for the function to be written for the database 

Note: I installed and used the extension to check that the coding style matches what is required and did not see any problems, please let me know if there is another way to check this!

### CODE REVIEW

File: frontend/index.html [hunk 1]
```diff
@@ -13,5 +13,32 @@
...
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json'
+          },
+          body: JSON.stringify(test_data)
+        }
+
+        fetch(url, req).then(res => res.json()).then(test_data => {console.log(test_data)})
+      }
+
```
Comment (nbriz) [inline frontend/index.html]: let's move this to it's own test JS file (that we can also use the linter to make sure it's conforming to our code style), something like `frontend/js/post-request-test.js`, then u can add it to the index page like this:
```html
<script src="js/post-request-test.js"></script>
```

**Review by nbriz — CHANGES_REQUESTED** [review]
hey @shayna-k solid PR from ur feature branch which follows our coding style (w/one tiny exception, the front-end test code see comment below).

there isn't a lot in the REST API endpoint yet though, i realize that this is b/c the backend team is still working on their part, but as i mentioned in Ishan's PR code review u can create a test file based on the [documentation Trisha posted here](https://github.com/net-art-uchicago/cta-file-sharing/issues/10#issuecomment-1315929803) in anticipation for when this will be ready.

create a file in the `backend` folder called `rest-api-test-new-data.js` which looks like this:

```js
const ex = { poem: 'a', date: 1, author: 'a', lat: 1, long: 1, route: 1 }

function validate (obj) {
  let pass = true
  for (const key in ex) {
    if (!obj[key]) pass = false
    else if (typeof obj[key] !== typeof ex[key]) pass = false
  }
  return pass
}

function addPoem (poem) {
  const passedValidation = validate(poem)
  if (passedValidation) return { success: true }
  else return {
    error: 'data failed validation',
    recieved: poem,
    expected: ex
  }
}

module.exports = addPoem
```

then u'll be able to import into ur REST API file like this:
```js
const addPoem = require('./rest-api-test-new-data.js')
```

which means u can then use it within the logic of ur endpoint like this:
```js
const status = addPoem(req.body)
```
then u can check the `status` variable, if u got a success message back from the test file than u should pass that success message along to the front end in ur  `res.json()` response, but if the `status` throws an error, u should let the front end know by passing that error along as well

**Comment by shayna-k** [comment]
@nbriz Ready for review:
- Added code to compare the posted poem to test data, and to add the poem
- Restructured the code to test posting poems 
- [included code to check a different poem to test errors]

Also, what is our requirement for documentation? I have some comments but not many explaining functionality as of now but can do so!

**Comment by nbriz** [comment]
looks great @shayna-k 
no specific requirements for documentation on this project (other than the documentation needed to communicate our plan to each other in the issues, so for example this issue: https://github.com/net-art-uchicago/cta-file-sharing/issues/14)

**Comment by nbriz** [comment]
i'm going to close this PR w/out merging it
there are some conflicts i had to resolve locally + via this [PR](https://github.com/net-art-uchicago/cta-file-sharing/pull/32)
u're still receiving credit for it though

**Pull request closed without merging**